fdm-core/src/field.ts

```diff
@@ -425,3 +426,185 @@ export async function updateField(
+                                        .b_id_harvestable_analysis,
+                                    harvestableAnalysisIds,
+                                ),
+                            )
+
+                        // Step 5: Delete from harvestable_analyses
+                        await tx
+                            .delete(schema.harvestableAnalyses)
+                            .where(
+                                inArray(
+                                    schema.harvestableAnalyses
+                                        .b_id_harvestable_analysis,
+                                    harvestableAnalysisIds,
+                                ),
+                            )
+                    }
+
+                    // Step 6: Delete from cultivation_harvesting
+                    await tx
+                        .delete(schema.cultivationHarvesting)
+                        .where(
+                            inArray(
+                                schema.cultivationHarvesting.b_lu,
+                                cultivationIds,
+                            ),
+                        )
+
+                    // Step 7: Delete from harvestables
+                    await tx
+                        .delete(schema.harvestables)
+                        .where(
+                            inArray(
+                                schema.harvestables.b_id_harvestable,
+                                harvestableIds,
+                            ),
+                        )
+                }
+
+                // Step 8: Delete from cultivation_starting, cultivation_ending and cultivations
+                await tx
+                    .delete(schema.cultivationStarting)
+                    .where(
+                        inArray(
+                            schema.cultivationStarting.b_lu,
+                            cultivationIds,
+                        ),
+                    )
+                await tx
+                    .delete(schema.cultivationEnding)
+                    .where(
+                        inArray(schema.cultivationEnding.b_lu, cultivationIds),
+                    )
+                await tx
+                    .delete(schema.cultivations)
+                    .where(inArray(schema.cultivations.b_lu, cultivationIds))
+            }
+
+            // Step 9: Get all soil analysis IDs for the field
+            const soilSamplings = await tx
+                .select({ a_id: schema.soilSampling.a_id })
+                .from(schema.soilSampling)
+                .where(eq(schema.soilSampling.b_id, b_id))
+
+            if (soilSamplings.length > 0) {
+                const soilAnalysisIds = soilSamplings.map((ss) => ss.a_id)
+
+                // Step 10: Delete from soil_sampling first
+                await tx
+                    .delete(schema.soilSampling)
+                    .where(inArray(schema.soilSampling.a_id, soilAnalysisIds))
```

Comment:
you actually delete everything. could you also make it inactive? or is that not desired?

Reply:
A trash option could be possible to add in the future or a permanent deletion after XX days